File: apps/hakka/Tests/CoreTests/SseFixtures.swift
import Foundation

/// Loads a pinned SSE wire fixture from the repo-shared `fixtures/sse/`
/// directory — the same transcripts the TypeScript presenter tests assert
/// against, so the two surfaces cannot drift apart.
enum SseFixtures {
    static func read(_ name: String) throws -> String {
        // …/hakka/apps/hakka/Tests/CoreTests/SseFixtures.swift → repo root
        let repoRoot = URL(fileURLWithPath: #filePath)
            .deletingLastPathComponent() // CoreTests
            .deletingLastPathComponent() // Tests
            .deletingLastPathComponent() // hakka (the package)
            .deletingLastPathComponent() // apps
            .deletingLastPathComponent()
        return try String(contentsOf: repoRoot.appendingPathComponent("fixtures/sse/\(name)"), encoding: .utf8)
    }
}

File: packages/hakka-browser/src/ui/llm/__tests__/sseFixtures.ts
import { readFileSync } from 'node:fs'
import { dirname, join } from 'node:path'
import { fileURLToPath } from 'node:url'

/**
 * Loads a pinned SSE wire fixture from the repo-shared `fixtures/sse/`
 * directory — the same transcripts the Swift presenter asserts against, so
 * the two surfaces cannot drift apart.
 */
const SSE_FIXTURES_DIR = join(
  dirname(fileURLToPath(import.meta.url)),
  '..',
  '..',
  '..',
  '..',
  '..',
  '..',
  'fixtures',
  'sse',
)

export function readSseFixture(name: string): string {
  return readFileSync(join(SSE_FIXTURES_DIR, name), 'utf-8')
}
